stactools_landsat/stactools/landsat/utils.py

```diff
@@ -0,0 +1,124 @@
+import datetime
+
+import dateutil
+import rasterio
+from pystac import Item, Link
+from rasterio import RasterioIOError
+from shapely.geometry import box, mapping, shape
+
+
+def _parse_date(in_date: str) -> datetime.datetime:
+    """
+    Try to parse a date and return it as a datetime object with no timezone
+    """
+    dt = dateutil.parser.parse(in_date)
+    return dt.replace(tzinfo=datetime.timezone.utc)
+
+
+def transform_mtl_to_item(metadata: dict) -> dict:
+    """
+    Handle USGS MTL as a dict and return a STAC dict.
+
+    This isn't implemented, because all the info we need
+    is already in the 0.7.0 STAC documents.
+    """
+    LANDSAT_METADATA = metadata["LANDSAT_METADATA_FILE"]
+    product = LANDSAT_METADATA["PRODUCT_CONTENTS"]
+    projection = LANDSAT_METADATA["PROJECTION_ATTRIBUTES"]
+    image = LANDSAT_METADATA["IMAGE_ATTRIBUTES"]
+    proessing_record = LANDSAT_METADATA["LEVEL2_PROCESSING_RECORD"]
+
+    scene_id = product["LANDSAT_PRODUCT_ID"]
+
+    xmin, xmax = float(projection["CORNER_LL_LON_PRODUCT"]), float(
+        projection["CORNER_UR_LON_PRODUCT"]
+    )
+    ymin, ymax = float(projection["CORNER_LL_LAT_PRODUCT"]), float(
+        projection["CORNER_UR_LAT_PRODUCT"]
+    )
+    geom = mapping(box(xmin, ymin, xmax, ymax))
+    bounds = shape(geom).bounds
+
+    # Like: "2020-01-01" for date and  "23:08:52.6773140Z" for time
+    dt = _parse_date(f"{image['DATE_ACQUIRED']}T{image['SCENE_CENTER_TIME']}")
+    created = _parse_date(proessing_record["DATE_PRODUCT_GENERATED"])
+
+    item = Item(id=scene_id, geometry=geom, bbox=bounds, datetime=dt, properties={})
+
+    # Common metadata
+    item.common_metadata.created = created
+    item.common_metadata.platform = image["SPACECRAFT_ID"]
+    item.common_metadata.instruments = [
+        i.lower() for i in image["SENSOR_ID"].split("_")
+    ]
+
+    # eo
+    item.ext.enable("eo")
+    item.ext.enable("view")
+    item.ext.enable("projection")
+
+    item.validate()
```

Comment:
The extensions are enabled, but the extension properties are not filled out - is that a TODO?

Reply:
Yep, not complete. Probably should delete the MTL function.

Reply:
That makes sense, if it's not useful to you, we'll wait until someone needs it to implement :-)